metadata-ingestion/src/datahub/emitter/rest_emitter.py

```diff
@@ -291,7 +294,8 @@ def emit_mcps(
         mcps: Sequence[Union[MetadataChangeProposal, MetadataChangeProposalWrapper]],
         async_flag: Optional[bool] = None,
     ) -> int:
-        logger.debug("Attempting to emit batch mcps")
+        if _DATAHUB_EMITTER_TRACE:
+            logger.debug(f"Attempting to emit MCP batch of size {len(mcps)}")
```

Comment:
We could remove such repeated conditionals, how about simply adding new function to the logging, i.e. in `logging_manager.py`:
```python
def trace(self: logging.Logger, message: str, *args: Any, **kwargs: Any):
    env = f'TRACE_{self.name.replace(".", "__").upper()}'
    if get_boolean_env_variable(env, False):
        self._log(logging.DEBUG, message, args, **kwargs)


logging.Logger.trace = trace
```
Then here we would be only calling `logger.trace(...)`. the variable to be exported in this case would be:
```shell
export TRACE_DATAHUB__EMITTER__REST_EMITTER=true
```
It would be quite universal piece of code

Reply:
Unfortunately mypy really doesn't like this type of patching

I agree with you that the code duplication isn't ideal, but it's better than having `type: ignore` everywhere. If there's an alternative that doesn't require that, I'm all ears